### PR TITLE
fix: Call to an undefined method Eloquent\Builder::restore

### LINKED ISSUE #1040: Call to an undefined method Eloquent\Builder::restore
- Larastan Version: 8.6.7
- `--level` used: max
- Pull request with failing test: https://github.com/nunomaduro/larastan/pull/1039

### Description
Larastan finds an issue when calling the method `Illuminate\Database\Eloquent\Builder::restore` of models that have the `SoftDeletes` trait.

### Laravel code where the issue was found

```php
<?php

class User extends Model
{
    use SoftDeletes;

    // ...
}

User::query()->onlyTrashed()->restore(); // Call to an undefined method Illuminate\Database\Eloquent\Builder<App\User>::restore().
```

### PR DESCRIPTION
**Pull request by ryo-imai-bit**
- [x] Added or updated tests
- [ ] Documented user facing changes
- [ ] Updated CHANGELOG.md

Fix #1040

**Changes**

Make sure Larastan find `Illuminate\Database\Eloquent\Builder::restore()`, and proper returnType.

### CODE REVIEW

**Comment by szepeviktor** [comment]
We have met #1059 in CI :(

**Comment by szepeviktor** [comment]
Could you please merge/rebase `master`?

File: tests/Features/Methods/Builder.php [hunk 1]
```diff
@@ -220,4 +220,10 @@ public function testValueWithQueryExpression()
         return User::with('foo')
             ->value(\Illuminate\Support\Facades\DB::raw('name'));
     }
+
+    /** @phpstan-return integer */
```
Comment (szepeviktor) [inline tests/Features/Methods/Builder.php:224]:
```suggestion
    /** @phpstan-return int */
```

Comment (ryo-imai-bit) [reply tests/Features/Methods/Builder.php]: Thanks! I fixed it.

**Comment by canvural** [comment]
Thank you!

**Pull request merged**